conjure-go-client/httpclient/metrics.go

```diff
@@ -107,37 +114,39 @@ func (h *metricsMiddleware) RoundTrip(req *http.Request, next http.RoundTripper)
 
 	var tags metrics.Tags
 	for _, tagProvider := range h.Tags {
-		tags = append(tags, tagProvider.Tags(req, resp)...)
+		tags = append(tags, tagProvider.Tags(req, resp, err)...)
 	}
 
 	metrics.FromContext(req.Context()).Timer(metricClientResponse, tags...).Update(duration / time.Microsecond)
 	return resp, err
 }
 
-func tagStatusFamily(_ *http.Request, resp *http.Response) metrics.Tags {
-	var tag metrics.Tag
+func tagStatusFamily(_ *http.Request, resp *http.Response, respErr error) metrics.Tags {
 	switch {
+	case isTimeoutError(respErr):
```

Comment:
Should we nest this under a check for `resp == nil` so we don't execute this code path on every request? From the docs on [RoundTrip](https://pkg.go.dev/net/http#RoundTripper) if a response is obtained then it must return `err == nil`, so we should be able to bypass this codepath in that case.
Alternatively we could pass through the `resp` and check it inside that func.

Reply:
I figured that in that case, the first statement of the function (`if respErr == nil`) would fire and return immediately with basically no cost. I can refactor but not sure it would make any difference

Reply:
Yeah that's fair